### DISCUSSION TITLE
graphql/codegen/plugins/typescript/typescript-vue-apollo-smart-ops

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#8622] [Docs Discussions]
# graphql/codegen/plugins/typescript/typescript-vue-apollo-smart-ops

GraphQL Code Generator Documentation

https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-vue-apollo-smart-ops

**Comment by harrixon** [comment]
Does this plugin support query on apollo cache?

I am getting GraphQL Document Validation Error on "Cannot query field "xxx" on type "Query".

Part of my schema
```graphql
type Group {
  id: String!
  members: [GroupMember]
}
```

Query to read / write on cache:
```typescript
const memberQuery = gql`
  query member($id: String!) {
    member(id: $id) {
      id
      name
    }
  }
`;
```

**Reply by charlypoly** [reply]
Hi @harrixon,

The error is linked to your GraphQL Schema implementation, not to the plugin.
Does your GraphQL Schema exposes a `member($id: String!): Member` query?

**Reply by harrixon** [reply]
Hi @charlypoly,
No it doesn't, I understand that the `member` query is not a valid query according to the schema. However, the query is used on vue-apollo's cache only, which I use it to mutate cached value.

**Reply by ikhidebright** [reply]
The vue-apollo-smart-ops has issues on vue3 as it depends on vue-class-component and vue-class-component is not supported on vue3, can you assist?

**Comment by ikhidebright** [comment]
The vue-apollo-smart-ops has issues on vue3 as it depends on vue-class-component and vue-class-component is not supported on vue3, can you assist?
